subsys/usb/class/hid/core.c

```diff
@@ -201,17 +215,36 @@ static void hid_int_in(u8_t ep, enum usb_dc_ep_cb_status_code ep_status)
 	hid_device.ops->int_in_ready();
 }
 
+#ifdef CONFIG_ENABLE_HID_INT_OUT_EP
+static void hid_int_out(u8_t ep, enum usb_dc_ep_cb_status_code ep_status)
+{
+	if (ep_status != USB_DC_EP_DATA_OUT ||
+	    hid_device.ops->int_out_ready == NULL) {
+		return;
+	}
+	hid_device.ops->int_out_ready();
+}
+#endif
+
 /* Describe Endpoints configuration */
 static struct usb_ep_cfg_data hid_ep_data[] = {
 	{
 		.ep_cb = hid_int_in,
-		.ep_addr = CONFIG_HID_INT_EP_ADDR
+		.ep_addr = CONFIG_HID_INT_IN_EP_ADDR,
+	},
+#ifdef CONFIG_ENABLE_HID_INT_OUT_EP
+	{
+		.ep_cb = hid_int_out,
+		.ep_addr = CONFIG_HID_INT_OUT_EP_ADDR,
+
 	}
+#endif
 };
 
 static void hid_interface_config(u8_t bInterfaceNumber)
 {
 	hid_cfg.if0.bInterfaceNumber = bInterfaceNumber;
+	hid_cfg.if0.bNumEndpoints = ARRAY_SIZE(hid_ep_data);
 }
```

Comment:
Minor nitpick, maybe it's a good idea to set `bNumEndpoints` in the structure definition to `0` so that it's clear that it is going to be updated later. 

Alternatively, as the value is known is known at compile time, it might be better to just do something like that:
```
+#ifdef CONFIG_ENABLE_HID_INT_OUT_EP
+               .bNumEndpoints = 2,
+#else
                .bNumEndpoints = 1,
+#endif
```

Reply:
We do not need more ifdefs here, ARRAY_SIZE or 0 are better options.